app/src/features/projects/components/ProjectLocationForm.tsx

```diff
@@ -32,10 +44,31 @@ export interface IProjectLocationFormProps {
  * @return {*}
  */
 const ProjectLocationForm: React.FC<IProjectLocationFormProps> = (props) => {
+  const classes = useStyles();
+
   const formikProps = useFormikContext<IProjectLocationForm>();
 
   const { values, touched, errors, handleChange, handleSubmit } = formikProps;
 
+  const [bounds, setBounds] = useState<any>([]);
+  const [geometry, setGeometry] = useState<Feature[]>([]);
+
+  const handleSpatialUpload = async (e: any) => {
+    const file = e.target.files[0];
+    const fileAsString = await file.text().then((xmlString: string) => {
+      return xmlString;
+    });
+    const domKml = new DOMParser().parseFromString(fileAsString, 'application/xml');
+    const geojson = kml(domKml);
+    const bboxCoords = bbox(geojson);
```

Comment:
What happens if I pick a non spatial file?

Do we need to wrap this in a try/catch and display some kind of error message if it fails?

Reply:
yeah for sure